src/subset.c

```diff
@@ -124,39 +124,62 @@ const char *check_idx(SEXP idx, int max, bool *anyNA_out, bool *orderedSubset_ou
   return NULL;
 }
 
-SEXP convertNegAndZeroIdx(SEXP idx, SEXP maxArg, SEXP allowOverMax)
+SEXP convertNegAndZeroIdx(SEXP idx, SEXP maxArg, SEXP allowOverMax, SEXP nomatchArg)
```

Comment:
the name of the new argument (`noMatchArg`) needs knowledge of where this function is being called from to understand. A better name might be `allowNA` or `removeNA` to more directly convey what it does, iiuc.